File: src/main/java/com/dafttech/terra/game/world/environment/WeatherRainy.java
package com.dafttech.terra.game.world.environment;

import com.badlogic.gdx.Gdx;
import com.dafttech.terra.TerraInfinita;
import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.game.world.World;
import com.dafttech.terra.game.world.entities.particles.ParticleRain;

public class WeatherRainy extends Weather {

    @Override
    public void update(World world, float delta) {
        Vector2 top = world.localPlayer().getPosition().sub(0, 0 + (Gdx.graphics.getHeight() / 2));

        if ((delta * 60f) * TerraInfinita.rnd().nextFloat() > 0.5f) {
            new ParticleRain(top.add((TerraInfinita.rnd().nextFloat() - 0.5f) * Gdx.graphics.getWidth(), 0), world).setVelocity(new Vector2(
                    (TerraInfinita.rnd().nextFloat() - 0.1f /* Rain direction */) * 25f, (TerraInfinita.rnd().nextFloat() + 5f) * 15f));
        }
    }

    @Override
    public float getWindSpeed(World world) {
        return 0;
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/world/environment/WeatherRainy.scala
package com.dafttech.terra.game.world.environment

import com.badlogic.gdx.Gdx
import com.dafttech.terra.TerraInfinita
import com.dafttech.terra.engine.vector.Vector2d
import com.dafttech.terra.game.world.GameWorld
import com.dafttech.terra.game.world.entities.particles.ParticleRain

class WeatherRainy extends Weather {
  override def update(delta: Float)(implicit world: GameWorld): Unit = {
    val top = world.localPlayer.getPosition - (0, Gdx.graphics.getHeight / 2)

    if (delta * 60f * TerraInfinita.rnd.nextFloat > 0.5f)
      new ParticleRain(
        top + ((TerraInfinita.rnd.nextFloat - 0.5f) * Gdx.graphics.getWidth, 0),
      ).setVelocity(Vector2d((TerraInfinita.rnd.nextFloat - 0.1 /* Rain direction */) * 25, (TerraInfinita.rnd.nextFloat + 5) * 15))
  }

  override def getWindSpeed(implicit world: GameWorld): Float = 0
}